src/webportal/src/app/job-submission/job-submission-page.jsx

```diff
@@ -0,0 +1,435 @@
+    getChecksum(name.slice(0, -2)) === name.slice(-2)
+  ) {
+    name = name.slice(0, -9);
+  }
+
+  let suffix = Date.now().toString(16);
+  suffix = suffix.substring(suffix.length - 6);
+  name = `${name}_${suffix}`;
+  name = name + getChecksum(name);
+  return name;
+}
+
+export const JobSubmissionPage = ({isSingle, setWizardStatus, yamlText}) => {
+  const [jobTaskRoles, setJobTaskRolesState] = useState([
+    new JobTaskRole({name: 'Default_Task_Role'}),
+  ]);
+  const [parameters, setParametersState] = useState([{key: '', value: ''}]);
+  const [secrets, setSecretsState] = useState([{key: '', value: ''}]);
+  const [jobInformation, setJobInformation] = useState(
+    new JobBasicInfo({
+      name: `${loginUser}_${Date.now()}`,
+      virtualCluster: 'default',
+    })
+  );
+  const [selected, setSelected] = useState(SIDEBAR_PARAM);
+  const [advanceFlag, setAdvanceFlag] = useState(false);
+  const [jobData, setJobData] = useState(new JobData());
+  const [extras, setExtras] = useState({});
+  const [jobProtocol, setJobProtocol] = useState(new JobProtocol({}));
+  const [loading, setLoading] = useState(true);
+
+  // Context variables
+  const [vcNames, setVcNames] = useState([]);
+  const [errorMessages, setErrorMessages] = useState({});
+
+  const setJobTaskRoles = useCallback(
+    (taskRoles) => {
+      if (isEmpty(taskRoles)) {
+        setJobTaskRolesState([new JobTaskRole({name: 'Task_role_1'})]);
+      } else {
+        setJobTaskRolesState(taskRoles);
+      }
+    },
+    [setJobTaskRolesState]
+  );
+
+  const setParameters = useCallback(
+    (param) => {
+      if (isEmpty(param)) {
+        setParametersState([{key: '', value: ''}]);
+      } else {
+        setParametersState(param);
+      }
+    },
+    [setParametersState]
+  );
+
+  const setSecrets = useCallback(
+    (secret) => {
+      if (isEmpty(secret)) {
+        setSecretsState([{key: '', value: ''}]);
+      } else {
+        setSecretsState(secret);
+      }
+    },
+    [setSecretsState]
+  );
+
+  const onSelect = useCallback(
+    (x) => {
+      if (x === selected) {
+        setSelected(null);
+      } else {
+        setSelected(x);
+      }
+    },
+    [selected, setSelected]
+  );
+
+  const setErrorMessage = useCallback(
+    (id, msg) => {
+      setErrorMessages((prev) => {
+        if (isEmpty(msg)) {
+          if (prev !== undefined && prev[id] !== undefined) {
+            const updated = {...prev};
+            delete updated[id];
+            return updated;
+          }
+        } else {
+          if (prev !== undefined && prev[id] !== msg) {
+            return {
+              ...prev,
+              [id]: msg,
+            };
+          }
+        }
+        return prev;
+      });
+    },
+    [setErrorMessages]
+  );
+
+  const contextValue = useMemo(
+    () => ({
+      vcNames,
+      errorMessages,
+      setErrorMessage,
+    }),
+    [vcNames, errorMessages, setErrorMessage]
+  );
+
+  useEffect(() => {
```

Comment:
Move it to after the `dockerInfo` update? We want this use effect to be execute after the useEffect in line `234`. (The bug we discussed yesterday) Maybe we need to create an other state such as `initialized` to make these useEffect executed as our expect

Reply:
change order in temp, and maybe further improved in release branch